File: 1854-maximum-population-year/1854-maximum-population-year.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        years = [0] * 2051
        for birth, death in logs:
            years[birth] += 1
            years[death] -= 1
        cur_pop = max_pop = year = 0
        for yr in range(1950, 2051):
            cur_pop += years[yr]
            if cur_pop > max_pop:
                max_pop = cur_pop
                year = yr
        return year
            

File: 1854-maximum-population-year/1854-maximum-population-year.js
/**
 * @param {number[][]} logs
 * @return {number}
 */
var maximumPopulation = function(logs) {
    const years = Array(2051).fill(0)
    for (const [birth, death] of logs) {
        years[birth]++
        years[death]--
    }
    let cur = 0, max = 0, res = 0
    for (let year=1950; year <= 2050; year++) {
        cur += years[year]
        if (cur > max) {
            max = cur
            res = year
        }
    }
    return res
};